packages/indexer/src/vcr/record.ts

```diff
@@ -0,0 +1,52 @@
+import type { Client, StreamDataResponse } from "@apibara/protocol";
+import { run, type Indexer } from "@apibara/indexer";
+import type { CassetteOptions, VcrConfig } from "./config";
+import { klona } from "klona/full";
+import path from "node:path";
+import fs from "node:fs/promises";
+
+export type CassetteDataType<TFilter, TBlock> = {
+  filter: TFilter;
+  messages: StreamDataResponse<TBlock>[];
+};
+
+export async function record<TFilter, TBlock, TRet>(
+  vcr: VcrConfig,
+  client: Client<TFilter, TBlock>,
+  indexer: Indexer<TFilter, TBlock, TRet>,
+  cassetteOptions: CassetteOptions,
+) {
+  const _indexer = klona(indexer);
```

Comment:
Same as the other comment, change the input `indexer` to something else so we use `indexer` in the function.